Composer/packages/server/src/models/connector/csharpBotConnector.ts

```diff
@@ -2,74 +2,142 @@
 // Licensed under the MIT License.
 
 import fs from 'fs';
-import Path from 'path';
+import { ChildProcess, spawn } from 'child_process';
 
-import axios from 'axios';
 import archiver from 'archiver';
-import FormData from 'form-data';
 
 import { BotProjectService } from '../../services/project';
 import { DialogSetting } from '../bot/interface';
+import { Path } from '../../utility/path';
 
 import { BotConfig, BotEnvironments, BotStatus, IBotConnector, IPublishHistory } from './interface';
-
+let runtime: ChildProcess | null = null;
 export class CSharpBotConnector implements IBotConnector {
-  private adminEndpoint: string;
+  public status: BotStatus = BotStatus.NotConnected;
   private endpoint: string;
-  constructor(adminEndpoint: string, endpoint: string) {
-    this.adminEndpoint = adminEndpoint;
+  constructor(endpoint: string) {
     this.endpoint = endpoint;
   }
 
-  public status: BotStatus = BotStatus.NotConnected;
+  private stop = () => {
+    shutdown();
+    this.status = BotStatus.NotConnected;
+  };
 
-  connect = async (_: BotEnvironments, __: string) => {
-    // confirm bot runtime is listening here
-    try {
-      await axios.get(this.adminEndpoint + '/api/admin');
-    } catch (err) {
-      throw new Error(err);
+  private buildProcess = async (dir: string): Promise<number | null> => {
+    return new Promise((resolve, reject) => {
+      const build = spawn(`pwsh ./Scripts/build_runtime.ps1`, {
+        cwd: dir,
+        detached: true,
+        shell: true,
+        stdio: ['ignore', 'ignore', 'inherit'],
```

Comment:
Why ignore the `stderr` stream?

Reply:
Because I find it's corresponding to stdin, stdout and stderr respectively, so I ignore both in and out streams, just keep the stderr. Do you think it's necessary to keep in&out stream?
![image](https://user-images.githubusercontent.com/21174180/70017963-96227980-15bf-11ea-9425-1e60c9e36c1a.png)

Reply:
You are right. I had remembered the order backwards.